common/connection.go

```diff
@@ -28,6 +28,19 @@ const wsWriteBufferSize = 1 << 20
 var _ EventEmitter = &Connection{}
 var _ cdp.Executor = &Connection{}
 
+// Each connection needs its own msgID. A msgID will be used by the
+// connection and associated sessions. When a CDP request is made to
+// chrome, it's best to work with unique ids to avoid the Execute
+// handlers working with the wrong response, or handlers deadlocking
+// when their response is rerouted to the wrong handler.
+type msgID struct {
+	id int64
+}
+
+func (m *msgID) new() int64 {
+	return atomic.AddInt64(&m.id, 1)
+}
```

Comment:
Do you think it would make sense to abstract this behind an interface such as:
```
type msgIDGenerator interface {
    newID() int64
}
```
So we would hold this type in both `Connection` and `Session` and guarantee there is no direct access to the underlying `id` var?